crates/ntx-builder/src/builder/block_producer.rs

```diff
@@ -0,0 +1,47 @@
+use std::net::SocketAddr;
+
+use miden_node_proto::generated::{
+    block_producer::api_client::ApiClient, requests::SubmitProvenTransactionRequest,
+};
+use miden_node_utils::tracing::grpc::OtelInterceptor;
+use miden_objects::transaction::ProvenTransaction;
+use miden_tx::utils::Serializable;
+use tonic::{Status, service::interceptor::InterceptedService, transport::Channel};
+use tracing::{info, instrument};
+
+use crate::COMPONENT;
+
+type InnerClient = ApiClient<InterceptedService<Channel, OtelInterceptor>>;
+
+/// Interface to the block producer's gRPC API.
+///
+/// Essentially just a thin wrapper around the generated gRPC client which improves type safety.
+#[derive(Clone, Debug)]
+pub struct BlockProducerClient {
+    inner: InnerClient,
+}
+
+impl BlockProducerClient {
+    /// Creates a new block producer client with a lazy connection.
+    pub fn new(block_producer_address: SocketAddr) -> Self {
+        let block_producer_url = format!("http://{block_producer_address}");
+        let channel =
+            tonic::transport::Endpoint::try_from(block_producer_url).unwrap().connect_lazy();
```

Comment:
Should this error be handled with a `Result`?

Reply:
Should be fine to unwrap here because we are converting from a `SocketAddr`, but forgot the safety comment. Added.